lib/fli/FliImpl.h

```diff
@@ -227,14 +346,45 @@ class FliTimerCache {
     FliImpl *impl;
 };
 
+class FliIterator : public GpiIterator {
+public:
+    enum OneToMany {
+        OTM_END = 0,
+        OTM_CONSTANTS,  // include Generics
+        OTM_SIGNALS,
+        OTM_REGIONS,
+        OTM_SIGNAL_SUB_ELEMENTS,
+        OTM_VARIABLE_SUB_ELEMENTS
+    };
+
+    FliIterator(GpiImplInterface *impl, GpiObjHdl *hdl);
+
+    virtual ~FliIterator() { };
+
+    Status next_handle(std::string &name, GpiObjHdl **hdl, void **raw_hdl);
+
+private:
+    void populate_handle_list(OneToMany childType);
+
+private:
+    static GpiIteratorMapping<int, OneToMany> iterate_over;      /* Possible mappings */
+    std::vector<OneToMany> *selected;                            /* Mapping currently in use */
+    std::vector<OneToMany>::iterator one2many;
+
+    std::vector<void *> m_vars;
```

Comment:
@stuarthodgson 
Sorry about that, no the handles don't need to be freed.  Getting the handles through the iterators is just like getting the handles through the "mti_find*()" function.  I think this a pointer's to the actual "handle" that the simulator uses and not a copy of the information.  The approach here is identical to the examples in the FLI documentation